### PR TITLE
Adding a Dns Helper script

### PR DESCRIPTION
**Pull request by xleliberty**
Hi, 
This PR add an option to provide a dns helper script usually needed to have a clean dns resolution on ubuntu clients. 

A good dns helper script his https://github.com/jonathanio/update-systemd-resolved
and i've successfully used it with your action.
The important thing is that the script have to be executable.

Please tell me if it's something interesting in your opinion. 
I've already forked and modified this action, but i felt like this is a useful addition.

Xavier

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,6 +8,7 @@ Github action for connecting to OpenVPN server.
 | Name | Description | Required |
 | --- | --- | --- | 
 | `config_file` | Location of OpenVPN client config file | yes |
+| `dnshelper_script_file` | Location of Dns helper script file | Optional |
```
Comment (kota65535) [inline README.md:11]: Maybe `dns_helper_script_file` is better?

File: src/main.js [hunk 2]
```diff
@@ -32,6 +33,19 @@ const run = () => {
     fs.writeFileSync('ta.key', tlsAuthKey)
   }
 
+  // client dsn Script Path
```
Comment (kota65535) [inline src/main.js:36]: typo (dsn -> DNS)

File: src/main.js [hunk 3]
```diff
@@ -32,6 +33,19 @@ const run = () => {
...
+  if (dnsHelperScriptFile.length > 0) {
+    if (!fs.existsSync(dnsHelperScriptFile)) {
+      throw new Error(`dns helper script file '${dnsHelperScriptFile}' not found`)
+    }
+    const githubWorkspacePath = process.env.GITHUB_WORKSPACE
+
+    fs.appendFileSync(configFile, 'script-security 2\n')
+    fs.appendFileSync(configFile, `up ${githubWorkspacePath}/${dnsHelperScriptFile}\n`)
```
Comment (kota65535) [inline src/main.js:44]: It is recommended to call `up-restart` here.
cf. https://github.com/jonathanio/update-systemd-resolved#openvpn-configuration

**Comment by kota65535** [comment]
Hi @xleliberty, thank you for using and contributing to this action!
I reviewed and added some comments, please check them.

**Comment by kota65535** [comment]
Close due to inactivity.

**Comment by ShedPlant** [comment]
I copied this PR, or something like it, for my GitHub Actions workflow on `ubuntu-20.04` connecting to a OpenVPN.

But found when I upgraded to `ubuntu-22.04` runner, it wasn't necessary anymore 😄 .

**Comment by ShedPlant** [comment]
FWIW I found this PR useful to implement my own workaround.

I have a GitHub Actions workflow running on `ubuntu-22.04` which uses `kota65535/github-openvpn-connect-action` to connect to an OpenVPN.

I found that the VPN connected successfully but private DNS resolution didn't work.

I got it working with this in my `.ovpn` file:

```
# DNS for private hosted zone
dhcp-option DNS <my server>
# https://github.com/jonathanio/update-systemd-resolved
# https://github.com/kota65535/github-openvpn-connect-action/pull/1
script-security 2
up /path/to/copy/of/update-systemd-resolved.bash
up-restart
down /path/to/copy/of/update-systemd-resolved.bash
down-pre
```

**Pull request closed without merging**